var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -261,30 +269,49 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     depends_on('mmg', when='+mmg')
     depends_on('parmmg', when='+parmmg')
     depends_on('tetgen+pic', when='+tetgen')
+    # hypre+/~fortran based on wheter fortran is enabled/disabled
+    depends_on('hypre+fortran', when='+hypre+fortran')
+    depends_on('hypre~fortran', when='+hypre~fortran')
     # Hypre does not support complex numbers.
     # Also PETSc prefer to build it without internal superlu, likely due to
     # conflict in headers see
     # https://bitbucket.org/petsc/petsc/src/90564b43f6b05485163c147b464b5d6d28cde3ef/config/BuildSystem/config/packages/hypre.py
-    depends_on('hypre@:2.13+mpi~internal-superlu~int64', when='@:3.8+hypre+mpi~complex~int64')
-    depends_on('hypre@:2.13+mpi~internal-superlu+int64', when='@:3.8+hypre+mpi~complex+int64')
-    depends_on('hypre@2.14:2.18.2+mpi~internal-superlu~int64', when='@3.9:3.13+hypre+mpi~complex~int64')
-    depends_on('hypre@2.14:2.18.2+mpi~internal-superlu+int64', when='@3.9:3.13+hypre+mpi~complex+int64')
-    depends_on('hypre@2.14:2.22.0+mpi~internal-superlu~int64', when='@3.14:3.15+hypre+mpi~complex~int64')
-    depends_on('hypre@2.14:2.22.0+mpi~internal-superlu+int64', when='@3.14:3.15+hypre+mpi~complex+int64')
-    depends_on('hypre@2.14:+mpi~internal-superlu~int64', when='@3.16:+hypre+mpi~complex~int64')
-    depends_on('hypre@2.14:+mpi~internal-superlu+int64', when='@3.16:+hypre+mpi~complex+int64')
-    depends_on('hypre@develop+mpi~internal-superlu+int64', when='@main+hypre+mpi~complex+int64')
-    depends_on('hypre@develop+mpi~internal-superlu~int64', when='@main+hypre+mpi~complex~int64')
+    depends_on('hypre@:2.13+mpi~internal-superlu~int64',
+               when='@:3.8+hypre+mpi~complex~int64')
```

Comment:
I don't think this reformat is required [right now - in this PR]. Best to revert this for now [to keep the diff specific to this feature].

Will have to simplify this code at some point [so can be reformatted at that time]

Reply:
Done.